### DISCUSSION TITLE
Rules: analog input into var?

### DISCUSSION THREAD

**Discussion by tsunfinger** [arendst/Tasmota#19625] [Q&A]
Probably a really stupid question:

How to save a value from analog input into a variable on rules (analog input isn't trigger)?

Tom

**Comment by barbudor** [comment]
Not sure what you mean with "analog input isn't trigger" but the only way is to use the analog input as a trigger
```
Rule1
  on Analog#A0 do var1 %value% endon
```

**Reply by sfromis** [reply]
This is the general way of getting values from the JSON published by Tasmota, as the JSON is generated continuously for sensor values, also when not published. Since you seem to be using ESP32, you also have the Berry function `tasmota.read_sensors` giving a string you can parse into a map using `json.load`. This is the same JSON as you see for publish.

**Comment by tsunfinger** [comment]
> ```
> Rule1
>   on Analog#A0 do var1 %value% endon
> ```
Works this way, but it's a trigger. It will update var1 every time A0 is changing. Am I wrong?
But I could place it in another rule and disable rule when value is needed.
Or just make a copy of var1 into another var if needed...
Thanks
Tom

**Reply by sfromis** [reply]
Not when that analog input is changed, but when the value is updated each sensor cycle, even if unchanged.

If you just want it once, you could use the command `Status 10` (`Backlog Status 10` if from a rule) with a rule like `on StatusSNS#ANALOG#A1 do var1 %value% endon` to pick up the value from the command JSON output.

Anyway, your other recent thread looked like you're using ESP32, where I'd much recommend to not do it with rules, but the stronger Berry language instead.
